examples/3d/render_to_texture.rs

```diff
@@ -0,0 +1,216 @@
+            transform: Transform::from_translation(Vec3::new(0.0, 0.0, 1.0)),
+            ..Default::default()
+        })
+        .insert(FirstPassCube)
+        .insert(RenderLayers::layer(1));
+
+    // light
+    // note: currently lights are shared between passes!
+    commands.spawn_bundle(PointLightBundle {
+        transform: Transform::from_translation(Vec3::new(0.0, 0.0, 10.0)),
+        ..Default::default()
+    });
+
+    // camera
+    let render_target = RenderTarget::Image(image_handle);
+    clear_color.insert(render_target.clone(), Color::WHITE);
+    let mut first_pass_camera = PerspectiveCameraBundle {
+        camera: Camera {
+            name: Some(FIRST_PASS_CAMERA.to_string()),
+            target: render_target,
+            ..Default::default()
+        },
+        transform: Transform::from_translation(Vec3::new(0.0, 0.0, 15.0))
+            .looking_at(Vec3::default(), Vec3::Y),
+        ..Default::default()
+    };
+    active_cameras.add(FIRST_PASS_CAMERA);
+
+    let camera_projection = &mut first_pass_camera.perspective_projection;
+    camera_projection.update(size.width as f32, size.height as f32);
+    first_pass_camera.camera.projection_matrix = camera_projection.get_projection_matrix();
+    first_pass_camera.camera.depth_calculation = camera_projection.depth_calculation();
+
+    commands
+        .spawn_bundle(first_pass_camera)
+        .insert(RenderLayers::layer(1));
+    // NOTE: omitting the RenderLayers component may cause a validation error:
+    //
+    // thread 'main' panicked at 'wgpu error: Validation Error
+    //
+    //    Caused by:
+    //        In a RenderPass
+    //          note: encoder = `<CommandBuffer-(0, 1, Metal)>`
+    //        In a pass parameter
+    //          note: command buffer = `<CommandBuffer-(0, 1, Metal)>`
+    //        Attempted to use texture (5, 1, Metal) mips 0..1 layers 0..1 as a combination of COLOR_TARGET within a usage scope.
+
+    let cube_size = 4.0;
+    let cube_handle = meshes.add(Mesh::from(shape::Box::new(cube_size, cube_size, cube_size)));
+
+    let material_handle = materials.add(StandardMaterial {
+        base_color_texture: Some(RENDER_IMAGE_HANDLE.typed()),
+        reflectance: 0.02,
+        unlit: false,
+        ..Default::default()
+    });
+
+    // add entities to the world
+    commands
+        .spawn_bundle(PbrBundle {
+            mesh: cube_handle,
+            material: material_handle,
+            transform: Transform {
+                translation: Vec3::new(0.0, 0.0, 1.5),
+                rotation: Quat::from_rotation_x(-std::f32::consts::PI / 5.0),
+                ..Default::default()
+            },
+            ..Default::default()
+        })
+        .insert(MainPassCube);
+
+    commands.spawn_bundle(PerspectiveCameraBundle {
+        transform: Transform::from_translation(Vec3::new(0.0, 0.0, 15.0))
+            .looking_at(Vec3::default(), Vec3::Y),
+        ..Default::default()
+    });
+}
+
+fn main() {
```

Comment:
As a matter of style, I tend to like opening up examples with the `main` function, and briefly walking the reader through the logical flow.

Reply:
Done.